core/variant/typed_array.h

```diff
@@ -46,10 +46,15 @@ class TypedArray : public Array {
 		_ref(p_array);
 	}
 	_FORCE_INLINE_ TypedArray(const Variant &p_variant) :
-			Array(Array(p_variant), Variant::OBJECT, T::get_class_static(), Variant()) {
+			TypedArray(Array(p_variant)) {
 	}
-	_FORCE_INLINE_ TypedArray(const Array &p_array) :
-			Array(p_array, Variant::OBJECT, T::get_class_static(), Variant()) {
+	_FORCE_INLINE_ TypedArray(const Array &p_array) {
+		set_typed(Variant::OBJECT, T::get_class_static(), Variant());
```

Comment:
```suggestion
	_FORCE_INLINE_ TypedArray(const Array &p_array) :
			TypedArray() {
```
Alternatively, to reduce duplication even further